src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -3181,6 +3191,30 @@ impl Coordinator {
         Ok(Self::send_immediate_rows(rows))
     }
 
+    /// WARNING, ENTERING SPOOKY ZONE 3.0
+    ///
+    /// This method is ALWAYS called within the context of a specialized `tracing` `Subscriber` (set as
+    /// the thread-local default `Dispatch` using `.with_subscriber` at call-sites.
+    ///
+    /// In general, this should be considered safe, but `tracing` has limitations that mean that
+    /// any `Span` created under the specialized `Subscriber` that _leaves_ this function will
+    /// almost assuredly cause a panic inside `tracing`. This is because `Span`s track the
+    /// `Subscriber` they were created under, but certain actions (like an ordinary `Span` exit)
+    /// will call a method on the _thread-local_ `Subscriber`, which may be backed with a different
+    /// `Registry`.
+    ///
+    /// At first glance, there is no obvious way this method leaks `Span`s, but ANY tokio
```

Comment:
OK, so IIUC the invariant is that any `.async` methods nested inside `~_optimizer_pipeline` methods need to be called with a `.with_subscriber(root_dispatch)` using a specifically passed `rood_dispatch`. BTW none of the other `~_optimizer_pipeline` has nested `.await` calls.

Reply:
yes!